File: java/Send.java
import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.Channel;

public class Send {

  private final static String EXCHANGE_NAME = "mike.demo";

  public static void main(String[] argv) throws java.io.IOException {
    ConnectionFactory factory = new ConnectionFactory();
    factory.setHost("localhost");
    Connection connection = factory.newConnection();
    Channel channel = connection.createChannel();

    channel.exchangeDeclare(EXCHANGE_NAME, "fanout");
    // String message = getMessage(argv);
    
    while (!Thread.interrupted()) {
      try {
        String message = "Rand is: " + Math.random();
        channel.basicPublish(EXCHANGE_NAME, "", null, message.getBytes());
        Thread.sleep(500);
      } catch (InterruptedException e) {
        Thread.currentThread().interrupt();
      }
    }

    channel.close();
    connection.close();
  }
}

File: python/send.py
#!/usr/bin/env python
import pika
import sys
import time

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()

channel.exchange_declare(exchange='mike.demo', type='fanout')

message = ' '.join(sys.argv[1:]) or "info: Hello World!"

while True:
    channel.basic_publish(exchange='mike.demo', routing_key='', body=message)
    time.sleep(.5)

connection.close()
